tests/drivers/build_all/sensor/src/generic_test.c

```diff
@@ -155,16 +156,17 @@ static void run_generic_test(const struct device *dev)
 
 		/* Set this iteration's expected values in emul for every supported channel */
 		for (size_t i = 0; i < iodev_read_config.count; i++) {
-			enum sensor_channel ch = iodev_all_channels[i];
+			struct sensor_chan_spec ch = iodev_all_channels[i];
```

Comment:
NIT consider changing the name to something like `ch_spec`, as there is already a `ch` variable of a different type(`enum sensor_channel`) in same function above(line 106).